bulldozer/merge.go

```diff
@@ -56,7 +57,18 @@ func MergePR(ctx context.Context, pullCtx pull.Context, client *github.Client, m
 			if err != nil {
 				return errors.Wrap(err, "failed to determine pull request body")
 			}
-			commitMessage = body
+
+			if opt.CommitDescriptionDelimiter != nil {
+				var rString = fmt.Sprintf(
+					`(?sm:(%s\s*)^(.*)$(\s*%s))`, *opt.CommitDescriptionDelimiter, *opt.CommitDescriptionDelimiter)
+				if m := regexp.MustCompile(rString).FindStringSubmatch(body); len(m) == 4 {
+					commitMessage = m[2]
+				} else {
+					commitMessage = body
+				}
+			} else {
+				commitMessage = body
```

Comment:
You can simplify this conditional by setting `commitMessage` to `body` initially and then overwriting the value if a delimiter is set and the regexp matches.

Reply:
yeah, this is one part of go that I am not fond of and always code functionally. Will change